contracts/vesting/src/vesting/account/vesting_account.rs

```diff
@@ -0,0 +1,206 @@
+use crate::contract::NUM_VESTING_PERIODS;
+use crate::errors::ContractError;
+use crate::traits::VestingAccount;
+use config::defaults::DENOM;
+use cosmwasm_std::{Coin, Env, Order, Storage, Timestamp, Uint128};
+
+use super::Account;
+
+impl VestingAccount for Account {
+    fn locked_coins(
+        &self,
+        block_time: Option<Timestamp>,
+        env: &Env,
+        storage: &dyn Storage,
+    ) -> Result<Coin, ContractError> {
+        // Returns 0 in case of underflow.
+        Ok(Coin {
+            amount: Uint128::new(
+                self.get_vesting_coins(block_time, env)?
+                    .amount
+                    .u128()
+                    .saturating_sub(
+                        self.get_delegated_vesting(block_time, env, storage)?
+                            .amount
+                            .u128(),
+                    )
+                    .saturating_sub(
+                        self.get_bonded_vesting(block_time, env, storage)?
+                            .amount
+                            .u128(),
+                    ),
+            ),
+            denom: DENOM.to_string(),
+        })
+    }
+
+    fn spendable_coins(
+        &self,
+        block_time: Option<Timestamp>,
+        env: &Env,
+        storage: &dyn Storage,
+    ) -> Result<Coin, ContractError> {
+        Ok(Coin {
+            amount: Uint128::new(
+                self.load_balance(storage)?
+                    .u128()
+                    .saturating_sub(self.locked_coins(block_time, env, storage)?.amount.u128()),
+            ),
+            denom: DENOM.to_string(),
+        })
+    }
+
+    fn get_vested_coins(
+        &self,
+        block_time: Option<Timestamp>,
+        env: &Env,
+    ) -> Result<Coin, ContractError> {
+        let block_time = block_time.unwrap_or(env.block.time);
+        let period = self.get_current_vesting_period(block_time);
+
+        let amount = match period {
+            // We're in the first period, or the vesting has not started yet.
+            0 => Coin {
+                amount: Uint128::new(0),
+                denom: DENOM.to_string(),
+            },
+            // We always have 8 vesting periods, so periods 1-7 are special
+            1..=7 => Coin {
+                amount: Uint128::new(self.tokens_per_period()? * period as u128),
+                denom: DENOM.to_string(),
+            },
+            _ => Coin {
+                amount: self.coin.amount,
+                denom: DENOM.to_string(),
+            },
+        };
+        Ok(amount)
+    }
+
+    fn get_vesting_coins(
+        &self,
+        block_time: Option<Timestamp>,
+        env: &Env,
+    ) -> Result<Coin, ContractError> {
+        Ok(Coin {
+            amount: Uint128::new(
+                self.get_original_vesting().amount.u128()
```

Comment:
Why are we converting those from `Uint128` to `u128`? Theres a `Sub` trait defined on `Uint128`  directly

Reply:
Its probably pre cosmwasm 1.0 code, I think you could not do it directly before, will take a look